File: Python/lab4/factory/LaptopFactory.py
from interfaces.LaptopProduct import LaptopProduct
from interfaces.PcProduct import PcProduct
from interfaces.PhoneProduct import PhoneProduct
from interfaces.Factory import Factory

# Імпорт конкретних класів GamingLaptop і OficeLaptop з відповідних модулів
from laptop.GamingLaptop import GamingLaptop
from laptop.OficeLaptop import OficeLaptop

class LaptopFactory(Factory):
    def createLaptop(self, type_of_laptop) -> LaptopProduct:
        if type_of_laptop == 'Gaming':
            return GamingLaptop()
        elif type_of_laptop == "Ofice":
            return OficeLaptop()
        else:
            return None

    def createPc(self, type_of_pc) -> PcProduct:
        return None

    def createPhone(self, type_of_phone) -> PhoneProduct:
        return None

File: Java/lab4/src/main/java/org/example/factory/LaptopFactory.java
package org.example.factory;

import org.example.interfaces.Factory;
import org.example.interfaces.LaptopProduct;
import org.example.interfaces.PcProduct;
import org.example.interfaces.PhoneProduct;
import org.example.laptop.GamingLaptop;
import org.example.laptop.OficeLaptop;

public class LaptopFactory implements Factory {
    public LaptopProduct createLaptop(String typeOfLaptop) {
        return switch (typeOfLaptop) {
            case "Gaming" -> new GamingLaptop();
            case "Ofice" -> new OficeLaptop();
            default -> null;
        };
    }

    @Override
    public PcProduct createPc(String typeOfPc) {
        return null;
    }

    @Override
    public PhoneProduct createPhone(String typeOfPhone) {
        return null;
    }

}
